### PR TITLE
Added a utility module to generate random strings

### PR DESCRIPTION
**Pull request by jasonneurohr**
##### SUMMARY
A new utility module to generate random strings. Useful for playbooks where a unique value is required to prevent clashes.

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
randomstring

### CODE REVIEW

File: lib/ansible/modules/utilities/randomstring.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: randomstring
+
+short_description: Generates a random string of the desired length with an optional prefix and/or suffix.
+
+version_added: "2.9"
```
Comment (Im0) [inline lib/ansible/modules/utilities/randomstring.py]: Should be `2.10`.

**Review by Im0 — CHANGES_REQUESTED** [review]
Only a minor change request.

I'd be interested to see if any pro's/con's for module vs plugin in this use case.

File: lib/ansible/modules/utilities/randomstring.py (hunk 1 above)
Comment (jasonneurohr) [reply lib/ansible/modules/utilities/randomstring.py]: Resolved now - thank you.

**Review by Im0 — APPROVED** [review]
LGTM

**Comment by jasonneurohr** [comment]
> 
> 
> Only a minor change request.
> 
> I'd be interested to see if any pro's/con's for module vs plugin in this use case.

No compelling reason one way or the other. I created an alternate implementation as a lookup plugin should that be the preference here. [PR #65026](https://github.com/ansible/ansible/pull/65026)

**Comment by Akasurde** [comment]
I would use the existing `password` plugin to generate random string - 

```yaml
- set_fact:
    random_pod_name: "{{ lookup('password', '/dev/null chars=ascii_lowercase,digits length=8') }}"
```

If you still feel this needs a separate plugin, please submit this to https://github.com/ansible-collections/community.general. Thanks for your contribution.

**Pull request closed without merging**